src/clj/puppetlabs/services/master/master_core.clj

```diff
@@ -873,6 +873,121 @@
          :headers {"Content-Type" "application/octet-stream"}
          :body (get-code-content environment code-id file-path)}))))
 
+(defn is-bolt-project?
+  [dir]
+  (or (fs/exists? (str dir "/bolt-project.yaml"))
+      (fs/exists? (str dir "/bolt.yaml"))
+      (fs/exists? (str dir "/Boltdir"))))
+
+(defn get-project-root
+  "Lookup a project by name and return the path where its project files are
+  located as a string."
+  [projects-dir project-name]
+  (let [boltdir-path (str projects-dir "/" project-name "/Boltdir")]
+    (if (fs/exists? boltdir-path)
+      boltdir-path
+      (str projects-dir "/" project-name))))
+
+(defn dirs-in-project-modulepath
+  "List all directories in a bolt project's modulepath. Returns a sequence of
```

Comment:
Is it really a sequence?? That's the one where order matters, right? I didn't think that's what was returned here..